dashboard/modules/job/utils.py

```diff
@@ -237,3 +238,43 @@ async def find_job_by_ids(
         return job
 
     return None
+
+
+async def find_jobs_with_job_ids(
+    gcs_aio_client: GcsAioClient,
+    job_info_client: JobInfoStorageClient,
+    job_ids: List[str],
+) -> Dict[str, JobDetails]:
+    """
+    Returns a dictionary of submission jobs with the given job ids, keyed by the job id.
+    """
+    driver_jobs, submission_job_drivers = await get_driver_jobs(gcs_aio_client)
+
+    # Filter down to the request job_ids
+    driver_jobs = {key: job for key, job in driver_jobs.items() if key in job_ids}
+    submission_job_drivers = {
+        key: job for key, job in submission_job_drivers.items() if job.id in job_ids
```

Comment:
Do we need to support job submission ids or just job ids here?

Reply:
this only needs to support job ids for this specific usecase. In the future, it may be useful to also support submission ids for other use cases but no point in implementing it now. I'll leave a comment clarifying the behavior.

Reply:
Ohh got it, I got confused by the job ID associated with the job submission and the actual job submission ID.